frontend/src/containers/DashboardHistory.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useMemo, useState } from "react";
+import { useParams } from "react-router-dom";
+import Breadcrumbs from "../components/Breadcrumbs";
+import Table from "../components/Table";
+import Spinner from "../components/Spinner";
+import AuditTrailService from "../services/AuditTrailService";
+import Search from "../components/Search";
+import {
+  useDashboard,
+  useDashboardHistory,
+  useDateTimeFormatter,
+} from "../hooks";
+
+interface PathParams {
+  dashboardId: string;
+}
+
+function DashboardHistory() {
+  const [filter, setFilter] = useState("");
+  const { dashboardId } = useParams<PathParams>();
+  const dateFormatter = useDateTimeFormatter();
+  const { dashboard } = useDashboard(dashboardId);
+  const { auditlogs } = useDashboardHistory(dashboard?.parentDashboardId);
+
+  const tableColumns = useMemo(
+    () => [
+      {
+        Header: "Action",
+        accessor: (props: any) => {
+          return AuditTrailService.getActionFromDashboardAuditLog(props);
+        },
+      },
+      {
+        Header: "Dashboard version",
+        accessor: "version",
+      },
+      {
+        Header: "Date",
+        accessor: "timestamp",
+        Cell: (props: any) => dateFormatter(props.value),
+      },
+      {
+        Header: "User",
+        accessor: "userId",
+      },
+    ],
+    []
+  );
+
+  const onSearch = (query: string) => {
+    setFilter(query);
+  };
+
+  if (!dashboard) {
+    return <Spinner label="Loading..." />;
```

Comment:
The Loading icon is appearing on the left of the page, can you center it with something like:

<Spinner className="text-center margin-top-9" label="Loading" />

Reply:
Good idea.